device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/InternalClient.java

```diff
@@ -1155,12 +1169,19 @@ private void verifyReportedProperties(Set<Property> reportedProperties) throws I
         }
     }
 
-    private void verifyReportedPropertiesAndVersion(Set<Property> reportedProperties, Integer version) throws IOException {
-        verifyReportedProperties(reportedProperties);
-
-        if (version != null && version < 0) {
-            throw new IllegalArgumentException("Version cannot be negative.");
+    private void verifyTwinOperationsAreSupported()
+    {
+        if (this.config.getProtocol() == HTTPS)
+        {
+            throw new UnsupportedOperationException(TWIN_OVER_HTTP_ERROR_MESSAGE);
```

Comment:
concern about this being a breaking change? 

also, the .NET sdk could also use an inspection of the flow. (it looks like only one method call throws an exception and even the exception message is outdated: https://github.com/Azure/azure-iot-sdk-csharp/blob/master/iothub/device/src/Transport/HttpTransportHandler.cs#L254)